src/models/templates/vaccine.ts

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable camelcase */
+/* copied from https://github.com/Notarise-gov-sg/api-notify/blob/master/src/model/notification/index.ts */
+import { Static, String, Literal, Array, Record, Union } from "runtypes";
+import { Vaccination } from "../../types";
+import { config } from "../../config";
+
+const channel_mode = Union(Literal("SPM"), Literal("SPMORSMS"));
+const delivery = Union(Literal("IMMEDIATE"), Literal("SCHEDULE"));
+
+const templateInputV4 = Record({
+  qrissuedts: String,
+  qrexpiryts: String,
+  name: String,
+  travelpassport: String,
+  vaccname: String,
+  vaccdate: String,
+  qrcode: String, // base64 qrcode content
+});
+
+const templateV4 = Record({
+  template_id: Literal("SAFETRAVEL-QR-NTF-04"),
+  template_input: templateInputV4,
+});
+const template_layout = Array(Union(templateV4));
+
+const cta = Record({
+  action_name: String,
+  action_url: String,
+  action_type: String,
+});
+const title = String;
+const sender_name = String;
+const sender_logo_small = String; // URL of logo
+const category = Literal("MESSAGES");
+const priority = Literal("HIGH");
+const uin = String; // NRIC or FIN
+const notification_req = Record({
+  uin,
+  channel_mode,
+  delivery,
+  template_layout,
+  cta: Array(cta),
+  title,
+  sender_name,
+  sender_logo_small,
+  category,
+  priority,
+});
+
+const spmPayloadDef = Record({
+  notification_req,
+});
+
+export type SpmPayload = Static<typeof spmPayloadDef>;
+
+type SpmTemplate = Static<typeof templateV4>;
+
+export const getSpmTemplateV4 = (
+  name: string,
+  qrCode: string,
+  passportNumber: string,
+  vaccinations: Vaccination[],
+  validFrom: string
+): SpmTemplate => {
+  const { validityInDays } = config.vaccination;
+  const expiryts = new Date(validFrom);
+  expiryts.setDate(expiryts.getDate() + validityInDays);
+
+  return {
+    template_id: "SAFETRAVEL-QR-NTF-04",
+    template_input: {
+      qrissuedts: validFrom,
+      qrexpiryts: expiryts.toISOString(),
+      name,
+      travelpassport: passportNumber,
+      vaccname: vaccinations[0].vaccineName,
```

Comment:
not sure about this. I think currently there is no use case of people having different brand/type for 1st/2nd dose.

but in the future, if people get different brand of vaccine how shld this be handled?
e.g. 1st dose brand A, then suddenly brand A kena ban for whatever reason, then that person got to take 2nd dose from another brand or even retake the whole 1st/2nd dose vaccine from another brand

Reply:
Hmm I was following [api-vaccine](https://github.com/Notarise-gov-sg/api-vaccine/blob/master/src/models/notarizedHealthCert/createNotarizedHealthCert/createUnwrappedVaccinationCert.ts#L27) which takes in an array of Vaccination.

But at this point in time, the notification only references the first item of the array.

Reply:
haha i asked in a telegram group chat, whether which vaccination record name to display (or even dun display). wait for their conclusion

Reply:
seems like no conclusion yet. Probably we will stick with this implementation first